source/models/index.py

```diff
@@ -1,29 +1,33 @@
+from genericpath import isfile
 import os
 from outputs.exceptions import *
 
 class Index:
-    def __init__(self, table, index_value):
+    def __init__(self, table, index_name, index_value = None):
         self.table = table
+        self.index_name = index_name
         self.index_value = index_value
 
     def serialize(self):
         os.makedirs(self.get_path(), exist_ok = True)
-    
+
     def get_path(self, index_value = None):
         if index_value == None:
-            return os.path.join(self.table.get_path(), os.path.join("indices" , self.index_value))
+            return os.path.join(self.table.get_path(), os.path.join("Indices" , self.index_name))
         else:
             return os.path.join(self.get_path(), self.index_value + ".txt")
 
-    def add_primary_key(self, primary_key, index_value):
-        old_primary_keys = set(self.get_primary_key(index_value))                
+    def add_primary_key(self, primary_key):
+        old_primary_keys = set(self.get_primary_key(self.index_value))                
         old_primary_keys.add(primary_key)
-        Index.__write_in_file(path = self.get_path(index_value), data = old_primary_keys)
-    
+        self.__write_in_file(data = old_primary_keys)
+
     def delete_primary_key(self, primary_key, index_value):
         old_primary_keys = self.get_primary_key(index_value)
-        old_primary_keys.remove(primary_key)
-        Index.__write_in_file(path = self.get_path(index_value), data = old_primary_keys)
```

Comment:
Same mistake you did before.
Don’t open same file in different method.
Create a method takes a list of primary_key

This method is doing the following 
- check if the list is empty, so delete the file
- Add overwrite the file with the list of primary key
- 
by other word use write_in_file method to do this and call it here

Reply:
can you explain more

Reply:
You should only write or delete file in the same method.
Rename write_in_file to update_in_file.

update_in_file should take an array of indices. 

If array is empty, delete the file
If not empty, overwrite the file by new values

Reply:
ok i got it 
solved